### ISSUE TITLE
Order grid page errors when pipe symbol (and possibly other special characters) appear in SKU

### ISSUE THREAD

**Issue by bhavyamodi29** [markshust/magento2-module-ordergrid#15]
### Preconditions
Magento Version : 2.4.6-p2
Module Version : 1.1.1
### Description
When we order a product which has pipe symbol ( “ | ” ) in its SKU , then we try to visit the order grid page , it shows an error as shown in screenshot-1.
### Screenshot-1
![image](https://github.com/markshust/magento2-module-ordergrid/assets/124759192/1e3b07ba-3d27-47c3-a332-a629c8b8c1b5)
### Steps To Reproduce
1. Clone clean shop.
2. Install “MarkShust_OrderGrid” module.
3. Go to the Backend->Catalog->Products.
3. Create any product with the pipe symbol ( “ | ” ) in its SKU ( e.g. SKU = Test | Product).
4. Visit frontend , then order that product.
5. Go to the Backend->Sales->Orders.
### Expected Result
When we try to visit the order grid page , it should display product’s details in the Order Items column on the order grid page as shown  in screenshot-2.
### Screenshot-2
![image](https://github.com/markshust/magento2-module-ordergrid/assets/124759192/dce51ea5-8788-4e9f-bf64-0589c61abcb3)
### Actual Result
When we try to visit the order grid page , it shows an error as shown in screenshot-1.
### Additional Information
The issue arises due to the use of the pipe symbol ( “ | ” ) as a delimiter in the explode function , which separates the SKU , Quantity , and Product Name , as shown in screenshot-3 . Consider using a different delimiter that is unlikely to appear in SKUs , such as a semicolon.
### File Name
markshust/magento2-module-ordergrid/Model/ResourceModel/Order/Grid/Collection.php
### Screenshot-3
![image](https://github.com/markshust/magento2-module-ordergrid/assets/124759192/db7d8071-4701-46e8-a0a3-a0e03ad1d7ed)

**Comment by erfanimani** [comment]
I have come across this issue a few times as well, but never ended up investigating it.

The pipe comes from an SQL concat function, I don't think changing it to a semicolon is a good idea — someone somewhere will run into the same problem. Because the pipe is used purely internal, instead it could perhaps be changed to something like `{ordergrid_separator}`.

**Comment by markshust** [comment]
I think it makes sense for you to be able to define the delimiter from the admin. Will this work? I don't think there ever will be a single character which works for everyone.

**Comment by erfanimani** [comment]
Hi @markshust, what do you think of my previous comment? From what I understand it doesn't need to be a single character (in which case there would be a string of characters that *would* work for everyone).

I don't think we should make the module more complex and get people to configure the module when it should be plug and play.

**Comment by kasperth** [comment]
@markshust I agree with @erfanimani on his approach. Keeping it it simple as possible.

**Comment by markshust** [comment]
Ah I don't know what I was thinking. I agree.

**Comment by kasperth** [comment]
@markshust When do you reckon looking into this? My colleague has en open PR with a possible solution.

**Comment by Nichollas96** [comment]
Hello, any updates on this topic?

**Comment by markshust** [comment]
Hi there,

Sorry, I didn't see or was notified of the PR. Here's a link to it for reference: https://github.com/markshust/magento2-module-ordergrid/pull/16

I'll try to check this out soon.

**Comment by markshust** [comment]
Merged in https://github.com/markshust/magento2-module-ordergrid/pull/16, this should now be resolved. Really appreciate the PR and reporting, and thanks for your patience!
